src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -74,8 +74,8 @@ public HttpResponse executeRequest(Request<?> request, Map<String, String> addit
             throws IOException, AuthFailureError {
         String url = request.getUrl();
         HashMap<String, String> map = new HashMap<>();
-        map.putAll(request.getHeaders());
         map.putAll(additionalHeaders);
+        map.putAll(request.getHeaders());
```

Comment:
I added comments and a conformance test of sorts. (It would be neat if there were some way to export the conformance test so it'd be easy to run against other HttpStack implementations, but I'm not sure how to do that short of creating a whole new volley-test library).